mobile/src/utils/functions.ts

```diff
@@ -0,0 +1,247 @@
+export const getAllStoredData = async () => {
+  const oid = await SecureStore.getItemAsync("oid");
+  const publicKey = await SecureStore.getItemAsync(oid + "_publicKey");
+  const privateKey = await SecureStore.getItemAsync(oid + "_privateKey");
+  const deviceRegistered = await SecureStore.getItemAsync(
+    oid + "_deviceRegistered"
+  );
+  const email = await SecureStore.getItemAsync("email");
+  const phoneNumber = await SecureStore.getItemAsync("phoneNumber");
+  //const deviceVerified = await SecureStore.getItemAsync(oid + "deviceVerified");
+  const RegistrationCompleted = await SecureStore.getItemAsync(
+    oid + "RegistrationCompleted"
+  );
+  console.warn("oid: ", oid);
+  console.warn("publicKey: ", publicKey);
+  console.warn("privateKey: ", privateKey);
+  console.warn("deviceRegistered: ", deviceRegistered);
+  console.warn("email: ", email);
+  console.warn("phoneNumber: ", phoneNumber);
+  console.warn("RegistrationCompleted: ", RegistrationCompleted);
+};
+
+export const setSecureStoreData = async (key: string, value: string) => {
+  await SecureStore.setItemAsync(key, value);
+};
+
+export const checkDeviceHasScreenLock = (
+  callback: (result: boolean | null, error: { message: string } | null) => void
+) => {
+  LocalAuthentication.getEnrolledLevelAsync()
+    .then((result) => {
+      const hasScreenLockMechanism =
+        result !== LocalAuthentication.SecurityLevel.NONE;
+      callback(hasScreenLockMechanism, null); // First argument is result, second is error
+    })
+    .catch((error) => {
+      callback(null, {
+        message: "Error checking device screen lock mechanism" + error.message,
+      });
+    });
+};
+
+export const performBiometricValidation = async (
+  callback: (
+    biometricCheckStatus: "success" | "error" | "checking",
+    error: { message: string } | null
+  ) => void
+) => {
+  const isHardwareSupported = await isBiometricCheckSupportedByDevice();
+  if (!isHardwareSupported) {
+    callback("error", {
+      message: "Biometric check is not supported",
+    });
+    console.warn("Biometric check is not supported", isHardwareSupported);
+  } else {
+    await biometricValidation()
+      .then((result) => {
+        console.warn("biometricValidation result", result);
+        if (result.result === "success") {
+          callback("success", null); // First argument is biometricCheckStatus, second is error
+        } else if (result.result === "checking") {
+          callback("checking", {
+            message: "Biometric check is not supported",
+          });
+        } else if (result.result === "error") {
+          callback("error", {
+            message: result.message || "Biometric check failed",
+          });
+        } else {
+          callback("error", {
+            message: "Biometric check failed",
+          });
+        }
+      })
+      .catch((error) => {
+        console.log("biometricValidation error", error);
```

Comment:
Should this not rather use the callback than just logging out? If not, should it not be `console.error` then

Reply:
Fixed.